webharness/harness.js

```diff
@@ -76,24 +101,69 @@ function wait_for_message() {
 }
 
 /*
+ * Return a promise for the next available message
+ * to come in on |socket|. If there is a queued
+ * message, resolves the promise immediately, otherwise
+ * waits for socket_message to receive one.
+ */
+function wait_for_admin_message() {
+  return wait_for_message(socket_messages_admin,socket_message_promises_admin);
+}
+
+/*
+ * Return a promise for the next available message
+ * to come in on |socket|. If there is a queued
+ * message, resolves the promise immediately, otherwise
+ * waits for socket_message to receive one.
+ */
+function wait_for_test_message() {
+  return wait_for_message(socket_messages_test,socket_message_promises_test);
+}
+
+/*
  * Send an object as a message on |socket|.
  */
-function send_message(data) {
-  socket.send(JSON.stringify(data));
+function send_message(room_to_send,data) {
+  socket.emit('message', {room: room_to_send, msg: JSON.stringify(data)});
 }
 
-function connect_socket() {
+/*
+* Send an object as a message on |socket|.
+*/
+function send_admin_message(data) {
+  send_message(adminroom,data);
+}
+
+/*
+* Send an object as a message on |socket|.
+*/
+function send_test_message(data) {
+  send_message(testroom,data);
+}
+
+
+function connect_socket(type) {
   var server = SpecialPowers.getCharPref("steeplechase.signalling_server");
   if (server.substr(server.length - 1) != "/") {
     server += "/";
   }
-  var room = SpecialPowers.getCharPref("steeplechase.signalling_room");
+  var adminroom = SpecialPowers.getCharPref("steeplechase.signalling_room");
```

Comment:
This variable over-writes the global var 'adminroom'. Not sure if that is a good idea.